tests/services/test_planning_svc.py

```diff
@@ -18,7 +20,7 @@
 ]
 
 test_string = '#{1_2_3} - #{a.b.c} - #{a.b.d} - #{a.b.e[filters(max=3)]}'
-target_string = '#{1_2_3} - 1 - 2 - 3'
+target_string = '0 - 1 - 2 - 3'
```

Comment:
With this PR, `#{1_2_3}` is a valid fact now and will be subject to substitution.

Reply:
Should `test_string` include invalid facts? Are there invalid facts?

Reply:
Hm, I guess my use of the word "valid" here is a bit off. The `#{1_2_3}` part of the `test_string` before just wasn't being interpreted as a fact, as it didn't have more than 2 periods in it. Now it is considered a fact. So it wasn't so much a "valid" vs "invalid" distinction.

Reply:
@bworrell  I hate to potentially rain on the parade here (the idea of trying to simplify fact rules is a fantastic one, as long as it gels with the plans for the ongoing fact refactor), but `#{1_2_3}` very much needs to be ignored by the planning service, or stuff will start breaking at the planner level, seemingly at random. This caused quite a little kerfuffle a while back, but the general idea is that some abilities use a construct similar to facts, but separated by `_` rather than `.` to reference fields that need to be filled in later. I think the term we've used internally for these is `global variable`. One example of this can be found here, as `origin_link_id`: https://github.com/mitre/stockpile/blob/master/data/abilities/execution/95727b87-175c-4a69-8c7a-a5d82746a753.yml. 

Have you accounted for the potential issues here? If not, we're going to need to patch this, unfortuantely.

Reply:
Could this be resolved by moving `link.apply_id()` before `_build_single_test_variant()` in all cases? ex: https://github.com/mitre/caldera/blob/32997d11d7fae96e569c18d51874204af1b77daf/app/utility/base_planning_svc.py#L83

As an aside, variables in commands are populated in multiple places: `#{origin_link_id}` in the link, `#{app.*}` (and maybe `#{payload}`?) in the ability, a bunch in the agent. Would it be better to do this all in a single place, maybe in the planning service?

Reply:
@ArtificialErmine what potential issues are you referring to? Are there test cases or documentation that refers to these behaviors or constraints? I referred to the RTD page on variables, facts, and abilities and tried to account for the existence of global variables in this code and didn't experience any issues in my testing (see PR for testing notes).

Reply:
^^^ We talked offline and while the `origin_link_id` case should be fine, other issues were identified where it's possible for a global/reserved variable to be replaced with a fact, which shouldn't be allowed. So there will be some work done over the next couple weeks to try and resolve that.